### ISSUE TITLE
Intro Help guide, new wizard to guide the user through the app features

### ISSUE THREAD

**Issue by supportreq** [owncloud/android#1104]
With the Material Design almost in place, we should really have an intro screen which introduces the features of our OwnCloud App something what dropbox or box or even Google Drive does..

@AndyScherzinger what are your thoughts??

___

TASKS

- [X] [GIT] Rebase legacy PR 
- [x] [DEV] Choose features to include and the order in which they should appear.
- [x] [DESIGN] Design some mockups. 
- [x] [DEV] Create new resources for the steps.
- [x] [DEV] Check old code and apply the changes needed. 
- [x] [DEV] Include flag to enable or disable it.
- [x] [DEV] Code review
- [X] [QA] Design test plan @jesmrec
- [X] [QA] Execute test plan @jesmrec
- [ ] [GIT] Merge branch

**Comment by AndyScherzinger** [comment]
Du you mean like a dialog with a view pager on the very first start up the app?

**Comment by supportreq** [comment]
yes

**Comment by supportreq** [comment]
I just checked the iOS app has it.. @rperezb can we have it in android as well

**Comment by AndyScherzinger** [comment]
Can someone provide some screenshots from the iOS app, to get a better picture of how this is intended to work and what kind of information should be shown?

**Comment by supportreq** [comment]
I will it to you in sometime

**Comment by supportreq** [comment]
It's an amazing animation where the owncloud logo shifts from one screen to another as an animation..

![image1](https://cloud.githubusercontent.com/assets/4680404/9403465/059cec1a-4805-11e5-9d0d-fc8966b9847a.png)
![image2](https://cloud.githubusercontent.com/assets/4680404/9403461/05976c36-4805-11e5-8ee6-84bd9967e809.png)
![image3](https://cloud.githubusercontent.com/assets/4680404/9403466/059da9de-4805-11e5-9ee8-c04acb10a9f2.png)
![image4](https://cloud.githubusercontent.com/assets/4680404/9403462/0599415a-4805-11e5-926e-c0540a9a3ff1.png)
![image5](https://cloud.githubusercontent.com/assets/4680404/9403463/059bea22-4805-11e5-9540-f18f98d6ed29.png)
![image6](https://cloud.githubusercontent.com/assets/4680404/9403464/059c3b08-4805-11e5-9eaa-7ffcfce8b888.png)

**Comment by AndyScherzinger** [comment]
Nice! Are the corresponding image resources available somewhere?

**Comment by AndyScherzinger** [comment]
iOS files for this can be found here: https://github.com/owncloud/ios/tree/develop/Owncloud%20iOs%20Client/SupportingFiles/Images/HelpGuide

**Comment by tobiasKaminsky** [comment]
This should occur on the very first time or everytime an account should be added?
(preferring the first one: only on first installation)

**Comment by AndyScherzinger** [comment]
I would also pledge for one time only on very first launch of the app and also for no animations

**Comment by rperezb** [comment]
Hey guides, on iOS we call it the help guide, this is shown the very first time the app is open (not per account basis)

I rename this issue to help guide

The splash screen is shown everytime the app is open and it´s an static image, the one used on iOS is: https://github.com/owncloud/ios/blob/develop/Owncloud%20iOs%20Client/SupportingFiles/Images/ImageBranding/Branding/Default-Portrait%402x~ipad.png

**Comment by supportreq** [comment]
@AndyScherzinger @tobiasKaminsky in case you guys can implement this will be awesome!!

**Comment by tobiasKaminsky** [comment]
We first need the approval of @jancborchardt @masensio or @davivel

**Comment by jancborchardt** [comment]
Yes, however this should be condensed down to fewer steps. 6 steps is too much and will annoy people on first startup. @nasli @rperezb where is the related iOS issue?

**Comment by rperezb** [comment]
@jancborchardt on iOS there are 6 slides too, @supportreq pasted them  https://github.com/owncloud/android/issues/1104#issuecomment-133315788

**Comment by jancborchardt** [comment]
@rperezb I know, however I said in a past issue that 6 are too many. I can’t remember now which, that’s why I’m asking @nasli if she remembers where I said that.

**Comment by nasli** [comment]
@jancborchardt, here it is owncloud/ios#404 and also related on desktop owncloud/client#3711

We added a skip button as you suggested if someone does not want to see the slides, and we agreed to keep the 6 slides. 

But, take the advantage now that has not yet started development and redesign it with any improvement ;)

**Comment by supportreq** [comment]
@tobiasKaminsky you working on this?

**Comment by tobiasKaminsky** [comment]
No. 
The artwork and what the images should display is something I cannot do very well, so this should be up to @jancborchardt.
When the images are ready @AndyScherzinger @LukeOwncloud @przybylski  or me can do the intro slideshow.

**Comment by AndyScherzinger** [comment]
I think the images are already there, see link to the ios repo. Imho we would need them in the different resolutions or at least in xxhdp since scaling down isn't so bad as having Android scaling up... Support it in the native resolution for all the sizes is probably a bad idea since it will at a lot ob KBs to the APK.

Besides that we would also need a video to have an idea about the animation that is mention (we I would probably drop for Android....)

And also a general design thing @jancborchardt the intro guide uses the pre oC 8.2 file type icons thus the intro screen images should get a refresh :grin:

**Comment by supportreq** [comment]
@AndyScherzinger @tobiasKaminsky I can help with the images.. please advice the resolution, and the points that need to be covered..

**Comment by przybylski** [comment]
Today in the morning, i put together something super super super super super super super hacky to get some initial look and feel, this is what i did:
https://drive.google.com/file/d/0B--v4KNhT4mVSThJcFV2WUVqeHc/view?usp=sharing

**Comment by nasli** [comment]
@przybylski  look nice! Better in an ownCloud Server ;)

 We will start including all the resources in svg files into the iOS project within new release.

Until then, you can get it from here https://s3.owncloud.com/owncloud/index.php/s/KCaxDdBUxG91bau

The IntroGuide.svg file includes all icons organized in separete layers for an easy way to edit them, I use Inskcape to create it, enjoy!

**Comment by supportreq** [comment]
@przybylski check https://github.com/federicoiosue/Omni-Notes intro guide.. it is simple but looks very powerful.. and it's parent library : https://github.com/PaoloRotolo/AppIntro

**Comment by przybylski** [comment]
very interesting, I wonder what will @jancborchardt say

**Comment by supportreq** [comment]
Check the demo app they have some amazing design : https://play.google.com/store/apps/details?id=paolorotolo.github.com.appintroexample

**Comment by AndyScherzinger** [comment]
Well, this depends on several things:
* #1220 needs to be merged first since the lib you mentioned is targeting 23 for compile and thus due to dependencies ownCloud have also have to else the build would be failing (so this work has already been done)
* question is if we want to introduce more dependencies and libraries for the singular use case of an intro or if if don't just implent it using a view pager like @przybylski did. Anything is is imho basically a styling thing. As soon as #1100 gets merge one could also simple add the fab on the bottom left (or anywhere else) as shown in the demo of this lib :)

Don't get me wrong on the lib, the example sure do look nice, but from a technical pov I would leave that.

**Comment by AndyScherzinger** [comment]
@nasli are you updating the intro graphics to incorporate the latest icon designs? :tada:

**Comment by tobiasKaminsky** [comment]
I would suggest to not only show the basic options of the owncloud app (as they are expected), but also go more into detail and show the newest features (once they are merged):
- remote thumbnails
- resized preview
- stop duplicating files
This should then also be adjusted for each new release and be shown the first start after an upgrade.

**Comment by przybylski** [comment]
Maybe showing new and significant features in a popup after upgrade? something like:

Did you know that now you can do A B and C?

Announcing bug fixes isn't appealing (at least not to me)

**Comment by tobiasKaminsky** [comment]
No, no bug fixes. This is just annoying.
I just thought that we can reuse this feature to not duplicate the work in order to show new features.

**Comment by supportreq** [comment]
@AndyScherzinger I agree dependencies will increase, with the issues being merged it should not be a tedious effort. Also @tobiasKaminsky agree we should show the new features, big ones like material design, FAB icons.. with a basic intro to the app like dropbox..

**Comment by tobiasKaminsky** [comment]
@przybylski So you are working on the technical implementation of this? :+1: 
@jancborchardt We now need a decision on what should be on 6 slides.
(Also we should keep in mind that (if we want to introduce new features) this needs to be changed every release, which means extra work.)

My idea is to show 6 initial basic functions if the app is started for the very first time (of course with a skip button).
On every release there is a "what's new?" slide with 2-3 new features.

**Comment by przybylski** [comment]
@tobiasKaminsky technical implementation is easy. I would really like @jancborchardt to specify at least minimal design (I do have an idea but I rather let professional to handle it ;) )
As for changes on every release I do have a plan how to ease the pain of changes :)

**Comment by tobiasKaminsky** [comment]
This is great :) Then we let this all up to you and @jancborchardt 
You are really a very big help :) Wish you were never gone ;)

**Comment by MTRichards** [comment]
@przybylski Welcome back!

**Comment by przybylski** [comment]
I haven't got too much time this week to work on this but current progress looks like this (texts and images are dummy, I would need some resource for android).

On first run:
<img width="451" alt="zrzut ekranu 2015-11-26 o 14 41 56" src="https://cloud.githubusercontent.com/assets/1734309/11424692/a6be1fc4-944f-11e5-8a77-d07d694a4b7c.png">

<img width="451" alt="zrzut ekranu 2015-11-26 o 14 42 19" src="https://cloud.githubusercontent.com/assets/1734309/11424700/abf1f9c0-944f-11e5-9397-49e1d0e8bba0.png">

On upgrade:
<img width="455" alt="zrzut ekranu 2015-11-26 o 15 04 34" src="https://cloud.githubusercontent.com/assets/1734309/11424716/c84d90ac-944f-11e5-8e3c-7f6cb3681bbc.png">


Forward and tick icon are taken from android so they probably don't match overall design.

From programmer standpoint adding screen is (almost) single line. You need to specify string resource id for title and description of feature and resource id to image (all of them are optional). And version in witch feature was introduced (in string form like "1.8.1") everything else would be shown automatically.

Eg:
`new FeatureItem(R.drawable.logo, R.string.welcome_feature_1_title, R.string.welcome_feature_1_text, "1.0.0", true)`
One thing to mention is that "Welcome to ownCloud" screen will be shown after this feature is introduced even on users who already have ownCloud installed.

**Comment by przybylski** [comment]
Code is ugly but accessible on branch https://github.com/owncloud/android/tree/whatsNew

**Comment by AndyScherzinger** [comment]
I can probably add the icon work needed and have a look at the code.
As for the background color I think @jancborchardt wanted us to move to #ffffff to have the same colors used by the web and the mobile apps.

**Comment by przybylski** [comment]
First I wanted to cleanup code a bit, any feedback from UX perspective will be valuable

**Comment by AndyScherzinger** [comment]
Is it okay for you to have other community devs committing changes to the branch or should we just provide any feedback in the issue?

**Comment by przybylski** [comment]
Commit, I have no objections.

But shouldn't #ffffff coloring be fixed across whole application instead of single activities?
I changed background to #ffffff here's preview:
<img width="453" alt="zrzut ekranu 2015-11-26 o 17 08 41" src="https://cloud.githubusercontent.com/assets/1734309/11427068/818bd00a-9460-11e5-9e98-03d8e019dfd5.png">

**Comment by AndyScherzinger** [comment]
It should, yes. I did that on a branch including a PR but this hasn't been merged to master yet.

**Comment by tobiasKaminsky** [comment]
Shouldn't the "skip" look like a button?
Currently "skip" and "next" look a bit lost there...

@przybylski Great work. Looking forward to have it in beta branch.

> One thing to mention is that "Welcome to ownCloud" screen will be shown after this feature is introduced even on users who already have ownCloud installed. 

Why is this so?
You could check for an existing account to distinguish between "update" or "new install", or?

**Comment by przybylski** [comment]
I was aiming for skip and next not look like a button, they are borderless, so they look more like integral part of activity.

> You could check for an existing account to distinguish between "update" or "new install", or?

That is actually a good idea :) in addition to the current mechanism.
Now it's currently storing a version code in a preference which indicate last version, when this changes, list of new features, if there are any, is shown. 

The whole idea is to show "Welcome to ownCloud!" once upon install, and every "What's new in ownCloud" only after upgrade.

**Comment by tobiasKaminsky** [comment]
it was just a minor idea, and I am not a UI expert, so I am fine with the borderless "buttons".

> The whole idea is to show "Welcome to ownCloud!" once upon install, and every "What's new in ownCloud" only after upgrade.

This is very good :+1:
Will this also work with beta version?

**Comment by przybylski** [comment]
It depends, what is the versioning with beta looks like?

**Comment by tobiasKaminsky** [comment]
20151126, 20151127, and so on

**Comment by tobiasKaminsky** [comment]
I have checked out your branch and tried it.
It really looks great!
But, is it possible to use swipe left/right to change between the screens? At least this was what I tried and I guess this is the most common way users try to use it.

**Comment by tobiasKaminsky** [comment]
Regarding the version number. I saw that you use eg. 1.8.0.
Maybe it is better to use the versionCode as in AndroidManifest: 10800000.
With that the beta version would work out of the box (as it is the same length: 20151125)

**Comment by przybylski** [comment]
version code is calculated from string version, putting version code is not programmer friendly ;)
I could add another field which would indicate beta version but i don't think that beta needs a "whats new" view.

Right/left swipe is not yet implemented.

**Comment by tobiasKaminsky** [comment]
Beta is not that important. Just thought that we can test the "what's new" screens already there.
This would also have the benefit that the beta user directly see the changes and can test the new features more, which is what we want.
As the field is a string value we can add the beta version "20151115" in it and test if it is parseable by the normal version "1.8.1" or if it is a 8-digit-value and then use this as beta?

The future idea would be to motivate the PR devs to also create a whats new screen for their PR...

**Comment by AndyScherzinger** [comment]
Do we really want what's new screens for PRs? Depending on the PR in case it is not a big feature but rather an optimization or even a bugfix I wouldn't open such a screen, except for a version which I would call a changelog. The actual whatsNew implementation from what I understand is rather a general introduction showcase.

**Comment by tobiasKaminsky** [comment]
No, not for every PR, but for every big PR.
In current beta I suggest:
- Resized Images
- switchListvsGridView (but only because the behaviour is inherited from the parent folder)
- multiSelect
- streaming
These are "complicated" PRs or things that need some explanation, I think.
All other PRs (like video thumbnail, cancel wlan on exist, material button/fab, upload when charging) are enhancements that are self explaining.

But for these 4 PRs I suggest to add a "whats new" screen and test this also in beta.

**Comment by przybylski** [comment]
Left/Right swipe is now on branch, if anyone could update images and descriptions I would really appreciate it

**Comment by tobiasKaminsky** [comment]
Great :+1: 
@AndyScherzinger maybe you can help with images and descriptions?

**Comment by AndyScherzinger** [comment]
Sure thing, I will have a look, but might need help from Jan for the vector images to create the images in the right sizes.

**Comment by przybylski** [comment]
@AndyScherzinger @jancborchardt any feedback?

**Comment by AndyScherzinger** [comment]
Hi @przybylski I haven't found the time yet to have a closer look, but it looks good from the screenshots though. Only thing I would change are minor graphical things, like having the circles without borders in a semi-transparent and dark blue and also have the right hand side arrow icon in the dark blue, but I can work on that later maybe even in a separate PR. So, looks good to me :)

**Comment by tobiasKaminsky** [comment]
Beside the (technical/visual) part we need someone who can do the images for the new PRs...?

**Comment by przybylski** [comment]
Please push images to whatsNew branch. It's currently in PR

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky for new images we probably need some help from @owncloud/designers :)

**Comment by AndyScherzinger** [comment]
I pushed the graphical changes to the branch, having changed some paddings/margins and the idicator visuals as well as the next/done icon (right, bottom) to a blue arrow and chec/blue-circle for the last action (kind of like the Google apps do it :smiley:).
The arrow has been implemented with a vector-xml and thus i bumped gradle to 1.5.0 unsure if this works for ant build processes (probably not, which will then force me to go back to png files, or the core team stops using ant, which is a legacy build system for android anyways :wink:)

![device-2016-01-06-171020](https://cloud.githubusercontent.com/assets/1315170/12147427/1ac612dc-b499-11e5-937b-e8a8f1107ed6.png)

![device-2016-01-06-171032](https://cloud.githubusercontent.com/assets/1315170/12147429/1db340dc-b499-11e5-8137-cfa81d99783c.png)

![device-2016-01-06-171040](https://cloud.githubusercontent.com/assets/1315170/12147432/20538892-b499-11e5-9a7f-677dcc313966.png)

![device-2016-01-06-171054](https://cloud.githubusercontent.com/assets/1315170/12147436/2306d828-b499-11e5-8499-d5d795008426.png)

![device-2016-01-06-171106](https://cloud.githubusercontent.com/assets/1315170/12147439/264e5eb6-b499-11e5-9ea3-0aef751fb3aa.png)

@davivel any plan or any chance to drop ANT support in the (near) future?

**Comment by AndyScherzinger** [comment]
@przybylski can't say what is is, but having the what's new screen open and turning the display off and on again seems to recalculate the idicator and the "blue/active) bubble is moved somewhere else. I guess there is some recalculation onResume (not sure).

**Comment by przybylski** [comment]
@AndyScherzinger thanks for graphics, i'll have a look on what could go wrong. Thanks again!

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger Looks awesome!

**Comment by enoch85** [comment]
Wow!

**Comment by AndyScherzinger** [comment]
@jancborchardt @nasli can anyone provide new (as in better and more adequate) feature images? 

The thing is, the images used are taken from the iOS repository and as you can see the icon work used in these images is pre 1.8.0 using the "old" icon set for the different mine/file-types but should rather use the new ones in my opinion. Does anyone have the original layered source images, the necessary image toolset (afaik inkscape) and time to do a overhaul? (which probably should then be made for all clients, so iOS and desktop too).

**Comment by AndyScherzinger** [comment]
@przybylski you are very welcome, it's great to see collaboration on PRs :tada: - one thing that might be subject to change is the vector part as mentioned before. :cry: 

What do you think @davivel ?
* *Pro* for the vector (+gradle): You can just change the color of the arrow in the xml file (for customer theming purposes) and everything is is handled by gradle's android plugin (generating png ressources for pre-v21).
* *Con* for the vector (+gradle): The vector implementation needs gradle to generate the pngs during build time which will lead to a broken ANT build (not tested though)

**Comment by AndyScherzinger** [comment]
One side note (sorry for the comment spamming):
The feature images are only present in the hdpi folder. Imho these should be stored only once but in the xhdpi or xxhdpi folder. I stored them in hdpi due to their resolution and don't have the vector original images to provide properly scaled ones.

In my opinion we should store them only in one resolution folder to not raise the App's file size to much and should only have the xxhdpi or xhdpi version since downscaling doesn't have such a blurring effect compared to upscaling (which Android will do).

**Comment by AndyScherzinger** [comment]
FYI I removed the gradle implication by adding the png files, but still I believe we should drop ANT support at some point even though it is used in the background build processes but thexyy should move on to gradle imho. Not being able to utilize Android native build system will probably get more painful in the future.

Anyway it should work now except the mentioned issue.

@przybylski One other effect I found is that the screen rotation resets the intro to step 1.

**Comment by przybylski** [comment]
@AndyScherzinger I am aware of rotation, I have never thought that someone would rotate device on whats new screen ;)
But I cannot replicate incorrect placing of current step indicator, but I guess I know where the problem is

**Comment by AndyScherzinger** [comment]
@przybylski one way would be to disallow rotation for the what's new screen - seems fine to me for this isolated case. :)

**Comment by przybylski** [comment]
neee it "only" needs to remember the state in onSaveState and onRestoreState, so thats doable

**Comment by tobiasKaminsky** [comment]
:+1: Rotation is needed as it is very inconvenience to force portrait modus.
(Imagine you use a tablet which is "always" in landscape and suddenly switch to portrait).

**Comment by jancborchardt** [comment]
I don't have the source files so pinging @nasli @rperezb @davivel about providing updated graphics with the new filetype icons etc. We need that for iOS as well anyway.

**Comment by nasli** [comment]
@przybylski, looks nice!! 
Summing up the resources we have:

For the new type of files icons:
https://github.com/owncloud/core/tree/90810cc6052b38ac03dd8f08a200c5928a355d20/core/img/filetypes

For the ios intro guide you can find it here:
https://github.com/owncloud/ios/tree/develop/svg_resources/IntroGuide

Until new versions, you can get it from here https://s3.owncloud.com/owncloud/index.php/s/KCaxDdBUxG91bau
but we don not have updated yet with the new icons.

Are you using some framework to create the animations? For the iOS we are using JazzHands and this openSource project now is also available for Android! It is called [SparkleMotion](https://github.com/IFTTT/SparkleMotion)

**Comment by tobiasKaminsky** [comment]
I thought these are static images? Are they not? Maybe you can provide a screencast of ios?
Then we/@przybylski can do it the same way.

**Comment by rperezb** [comment]
@tobiasKaminsky you may check it: https://s3.owncloud.com/owncloud/index.php/s/KCaxDdBUxG91bau 
There, you may fine¡d a video demo of the help intro

**Comment by nasli** [comment]
@tobiasKaminsky, some images are static but the cloud is animated. 
I just updated the images with the new filetypes
![newfiletypes](https://cloud.githubusercontent.com/assets/6570505/12643217/1b7f773a-c5bb-11e5-9fd1-294ffc074dcc.png)
![newfiletypesend](https://cloud.githubusercontent.com/assets/6570505/12643225/2bb8942e-c5bb-11e5-8b98-3a1e65439b3b.png)

the resources updated here https://s3.owncloud.com/owncloud/index.php/s/KCaxDdBUxG91bau

**Comment by jancborchardt** [comment]
@nasli that looks pixelated. Did you use the proper SVG icons?

**Comment by nasli** [comment]
@jancborchardt , these are just screenshot to show the new look.

In the resources folder you can find all the .png that we are going to use, and the IntroGuide.svg is a file where all views are separated by layers for easy import and scale to any size needed

**Comment by przybylski** [comment]
Fortunately android studio can transform SVG to correctly sized png resources

**Comment by tobiasKaminsky** [comment]
@przybylski 
What is needed to get this running in beta?
As I said earlier this would be great for every new release to show the recent big changes.
For instance I would like to test it with the new feature "pinch zoom" #1402.

**Comment by przybylski** [comment]
@tobiasKaminsky now on whatsNew branch in FeatureList.java file you can add features to be shown in beta. I haven't tested it but it should work ;) I'll fix it if needed. One thing to notice is in setup.xml there is a bool flag which indicates are we in beta or not

**Comment by tobiasKaminsky** [comment]
@przybylski I have checked out this branch and with a clean install I first get the intro screen prior to the server login. But after the server login I also get the whatsNew screens.
I think the first appearance is the correct.

**Comment by przybylski** [comment]
@tobiasKaminsky yes, i know about this bug I haven't yet investigated it

**Comment by tobiasKaminsky** [comment]
![whatsnewpinch](https://cloud.githubusercontent.com/assets/5836855/12865837/a5e75df8-ccb8-11e5-8637-4491a1acf8f3.png)

This would be the introduction of the pinchZoom feature.
But maybe the images are a little bit to small?

![whatsnewpinch1](https://cloud.githubusercontent.com/assets/5836855/12865861/56ebf5e6-ccb9-11e5-9667-d927c9435a52.png)

I really like this feature and hope it will also be used for every official release to show the 2-3 main changes.

**Comment by tobiasKaminsky** [comment]
It would be great if the featureItem would move while swiping, just like it is done in the ImageAdapter.

**Comment by jancborchardt** [comment]
@tobiasKaminsky the intro should not feature these detail actions. In fact it’s too long as it is and I would like it to ideally be just 3 slides. (As mentioned in the iOS issue about it before.) Only the really important stuff. Like Sync, Share, Instant Upload.

**Comment by przybylski** [comment]
@jancborchardt those basic features (instant upload, sync & share) will be presented on first install.
Additional features will be shown only on upgrade.

**Comment by jancborchardt** [comment]
Ok, but still I think this is only warranted for big features. And the screenshot above shows 6 dots / slides. ;)

**Comment by AndyScherzinger** [comment]
@jancborchardt btw the original screenshots of the iOS app has 6 dots too :wink:

**Comment by jancborchardt** [comment]
> In fact it’s too long as it is and I would like it to ideally be just 3 slides. (As mentioned in the iOS issue about it before.)

;)

**Comment by davigonz** [comment]
Taking into account the current status of the app, I would use 5 different steps for this wizard following this order:

1. Manage all your files
2. Share your files
3. Multiaccount
4. Camera uploads
5. Video streaming

**Comment by jesmrec** [comment]
Order is fine for me. 5 are a good number, so a longer wizard will make users to skip it.

**Comment by jesmrec** [comment]
About 1.

I miss the word `Sync`, `Syncing`, or `Keep on sync` anywhere. 

About 3.

`Connect to all your clouds` is a bit confusing, so it sounds to "external storage" like the core feature. I'd use: `Connect to all your ownClouds` or `Connect to all your servers`, `Manage many accounts` or something similar.

**Comment by michaelstingl** [comment]
@ChrisEdS Please review and provide feedback

**Comment by pmaier1** [comment]
Looks very nice so far but I have to ask about the intention/benefit. Should this be a feature overview to make the user aware of the nice features to captivate to explore those? Or does a user maybe know about the features from e.g. the app description on the appstore and actually needs some guidance on how to use them? How do I upload/move/copy/rename/delete a file? How can I share with another ownCloud user? Oh, I can generate links to files? And there's an option to forward them to mail or my favorite IM? How does the app integrate with Android features? What do I need to do for that the app uploads pictures/videos that I take?

**Comment by davigonz** [comment]
@pmaier1 Thanks for your feedback, this is an introduction to all the features included in the Android app, more visual and accessible than the description provided in Google PlayStore, we have a similar introduction in the iOS app as well. 

This intro is going to appear not only after first install but in updates, to show the new features included in that specific update, so that our users are aware about the potential of the app and about new stuff just included in the app.

About the detailed explanation for each feature, is something to be included in our [Android manual](https://doc.owncloud.org/android/).

**Comment by davigonz** [comment]
> I miss the word Sync, Syncing, or Keep on sync anywhere.

iOS wizard has a specific step with the message "Your private files synced anywhere" but it doesn't include Video streaming step. Including those words would be great but I don't know where exactly 😕

>  Connect to all your clouds is a bit confusing, so it sounds to "external storage" like the core feature. I'd use: Connect to all your ownClouds or Connect to all your servers, Manage many accounts or something similar.

What about a mix between two of your suggestions, "Connect to all your ownCloud servers"?

**Comment by jesmrec** [comment]
> iOS wizard has a specific step with the message "Your private files synced anywhere" but it doesn't include Video streaming step. Including those words would be great but I don't know where exactly

An option: in 1. change the title for `Manage all your synced files`. It is not direct, but at least we are telling the users that they can sync + manage.

> What about a mix between two of your suggestions, "Connect to all your ownCloud servers"?

sounds good!

**Comment by pmaier1** [comment]
Sorry, but I don't see much benefit in a firstrun/update wizard that tells the user that they can manage and share files, stream videos, do camera uploads but does not tell them how. Don't get me wrong - it looks nice - but how would it help the user?

In current state this is not more than a high-level feature list that the user needs to dismiss 5 times.
While this is of course better than nothing as an average user I would want to get answers to the questions in https://github.com/owncloud/android/issues/1104#issuecomment-365563086. Asking users to read a manual for an app sounds like 1996 :D

**Comment by jesmrec** [comment]
The welcome wizard is a way to welcome new uses to the app. Of course, those who download the app know what is it for, but, perhaps, not all of them know that the app handles with videostreaming, for example. The welcome wizard is a feature included in the iOS app from long ago, and it is a simple way to introduce basic features that you can skip with a simple click if not interested in. Many apps of all kinds use a welcome wizard and i think this is not something not-valuable. 

If the user needs further information about the app, in settings view we have a `Help` section that points to a web where different options are suggested (this has to be reviewed).

**Comment by davigonz** [comment]
> Sorry, but I don't see much benefit in a firstrun/update wizard that tells the user that they can manage and share files, stream videos, do camera uploads but does not tell them how. Don't get me wrong - it looks nice - but how would it help the user?

This is a pretty well-known feature in many mobile apps, the user needs an overview with the app features to know at a glance what the app is able to do.

> While this is of course better than nothing as an average user I would want to get answers to the questions in #1104 (comment). Asking users to read a manual for an app sounds like 1996 :D

> If the user needs further information about the app, in settings view we have a Help section that points to a web where different options are suggested (this has to be reviewed).

I agree with you @pmaier1 that we would need a better approach to answer the questions in https://github.com/owncloud/android/issues/1104#issuecomment-365563086. 

As @jesmrec says, we have a `Help` section in Android app which is basically a webview with the info contained in [https://owncloud.com/mobile/help](https://owncloud.com/mobile/help) but in my opinion, the optimal solution would be to point `Help`section to [https://doc.owncloud.org/android/android_app.html](https://doc.owncloud.org/android/android_app.html) , which contains the answers to https://github.com/owncloud/android/issues/1104#issuecomment-365563086 questions and is responsible so it will be adapted properly to a mobile device screen.
